### ISSUE TITLE
Changelog: Pull request merge queue (public beta)

### ISSUE THREAD

**Issue by yuhattor** [yuhattor/hattori.dev#133]
New Changelog: [Pull request merge queue (public beta)](https://github.blog/changelog/2023-02-08-pull-request-merge-queue-public-beta)

```md
# Blog Contents

## Title
Pull request merge queue (public beta)

## Link
https://github.blog/changelog/2023-02-08-pull-request-merge-queue-public-beta

## Published date
Wed, 08 Feb 2023 17:03:38 +0000

## Author
Kevin Duck

## Description
Today we are announcing the public beta of pull request merge queue for repos on GitHub Enterprise Cloud and open source organizations! 🎉

Merge queue helps increase velocity in software delivery by automating pull request merges into your busiest branches.

Before merge queue, developers were often required to update their pull request branches prior to merging to ensure their changes wouldn't break the main branch when merged. Each update resulted in a fresh round of continuous integration (CI) checks that would have to finish before the developer could attempt to merge. If another pull request got merged, every developer would have to go through the process again.

Merge queue automates this process by ensuring each pull request queued for merging is built with the pull requests ahead of it in the queue.

Queueing a pull request to merge

If your pull request targets a branch that uses merge queue, instead of merging your pull request directly when it meets the requirements to merge, you will add it to the queue by clicking Merge when ready from the pull request page or from GitHub Mobile.

The queue then creates a temporary branch that contains the latest changes from the base branch, the changes from other pull requests already in the queue, and the changes from your pull request. CI then starts, with the expectation that all required status checks must pass before the branch (and the pull requests it represents) are merged.

If a queued pull request has merge conflicts or fails any required status check, it is automatically removed from the queue when it reaches the front, and a notification is sent. Once the problem is resolved, it can be added back to the queue.

Learn more about merging a pull request with merge queue from the pull request page. You can also queue your pull request on the go using the beta version of GitHub Mobile from iOS TestFlight or Google Play (Beta) !

Viewing the queue

Always know where you are in the queue.

The queue details page, which can be accessed from the Branches page or pull request page, shows the pull requests in the queue and status for each, including the required status checks and estimated time to merge. It also shows how many pull requests have been merged and the trend over the last 30 days.

Depending on your permissions, you can also remove a pull request from the queue or clear the queue from this page.

Getting started

Merge queue can help improve overall velocity and avoid manual branch updates that impact developer productivity. Learn more about how to enable merge queue on your busiest branches.

We want to hear from you on how we can improve merge queue! Join the conversation in the merge queue public beta discussion .

## Raw Content
<p>Today we are announcing the public beta of pull request merge queue for repos on GitHub Enterprise Cloud and open source organizations! <g-emoji fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f389.png?v8" alias="tada">&#127881;</g-emoji></p>
<p>Merge queue helps increase velocity in software delivery by automating pull request merges into your busiest branches.</p>
<p><img decoding="async" src="https://i0.wp.com/user-images.githubusercontent.com/2503052/217027654-f570fb25-092d-476e-b6f5-0b31b8514662.png?ssl=1" alt="Pull request merge queue" data-recalc-dims="1"></p>
<p>Before merge queue, developers were often required to update their pull request branches prior to merging to ensure their changes wouldn&#039;t break the main branch when merged. Each update resulted in a fresh round of continuous integration (CI) checks that would have to finish before the developer could attempt to merge. If another pull request got merged, every developer would have to go through the process again.</p>
<p>Merge queue automates this process by ensuring each pull request queued for merging is built with the pull requests ahead of it in the queue.</p>
<h2 id="queueing-a-pull-request-to-merge" id="queueing-a-pull-request-to-merge" >Queueing a pull request to merge<a href="#queueing-a-pull-request-to-merge" class="heading-link pl-2 text-italic text-bold" aria-label="Queueing a pull request to merge"></a></h2>
<p>If your pull request targets a branch that uses merge queue, instead of merging your pull request directly when it meets the requirements to merge, you will add it to the queue by clicking <strong>Merge when ready</strong> from the pull request page or from GitHub Mobile.</p>
<p><img decoding="async" src="https://i0.wp.com/user-images.githubusercontent.com/2503052/217023882-cf4bf568-af5d-469e-882a-14f1837add72.gif?ssl=1" alt="Queue to merge" data-recalc-dims="1"></p>
<p>The queue then creates a temporary branch that contains the latest changes from the base branch, the changes from other pull requests already in the queue, and the changes from your pull request. CI then starts, with the expectation that all required status checks must pass before the branch (and the pull requests it represents) are merged.</p>
<p>If a queued pull request has merge conflicts or fails any required status check, it is automatically removed from the queue when it reaches the front, and a notification is sent. Once the problem is resolved, it can be added back to the queue.</p>
<p>Learn more about <a href="https://docs.github.com/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request-with-a-merge-queue">merging a pull request with merge queue</a> from the pull request page. You can also queue your pull request on the go using the beta version of GitHub Mobile from <a href="https://testflight.apple.com/join/NLskzwi5">iOS TestFlight</a> or <a href="https://play.google.com/apps/testing/com.github.android">Google Play (Beta)</a>!</p>
<h2 id="viewing-the-queue" id="viewing-the-queue" >Viewing the queue<a href="#viewing-the-queue" class="heading-link pl-2 text-italic text-bold" aria-label="Viewing the queue"></a></h2>
<p>Always know where you are in the queue.</p>
<p>The queue details page, which can be accessed from the Branches page or pull request page, shows the pull requests in the queue and status for each, including the required status checks and estimated time to merge. It also shows how many pull requests have been merged and the trend over the last 30 days.</p>
<p><img decoding="async" src="https://i0.wp.com/user-images.githubusercontent.com/2503052/217023511-13c729f7-7458-4ee3-8f7c-be5a7cf06f99.png?ssl=1" alt="Merge queue details page" data-recalc-dims="1"></p>
<p>Depending on your permissions, you can also remove a pull request from the queue or clear the queue from this page.</p>
<h2 id="getting-started" id="getting-started" >Getting started<a href="#getting-started" class="heading-link pl-2 text-italic text-bold" aria-label="Getting started"></a></h2>
<p>Merge queue can help improve overall velocity and avoid manual branch updates that impact developer productivity. Learn more about how to <a href="https://docs.github.com/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue">enable merge queue</a> on your busiest branches.</p>
<p>We want to hear from you on how we can improve merge queue! Join the conversation in the <a href="https://github.com/community/community/discussions/46757">merge queue public beta discussion</a>.</p>

```
